### ISSUE TITLE
Bug: `require-alt-text` does not report `<img>` when "alt" appears inside another attribute's value

### ISSUE THREAD

**Issue by minseonkkim** [eslint/markdown#741]
### Environment

ESLint version: v10.10.0
@eslint/markdown version: v8.0.3
Node version: v24.21.0
npm version: v11.19.0
Operating System: win32 10.0.26200

### Which language are you using?

commonmark

### What did you do?

<details>
<summary>Configuration</summary>

```js
import { defineConfig } from "eslint/config";
import markdown from "@eslint/markdown";

export default defineConfig([
   {
      files: ["**/*.md"],
      plugins: { markdown },
      language: "markdown/commonmark",
      extends: ["markdown/recommended"],
   },
]);
```
</details>

```md
# Title

<img src="a.png">

<img src="b.png" title="an alt">

<img src="c.png" class="no alt">

<img src="d.png" data-caption="Photo alt">

<img src="e.png" alt="Logo">
```

I ran `npx eslint test.md`.

### What did you expect to happen?

`markdown/require-alt-text` should report four images: `a.png`, `b.png`, `c.png`, and `d.png`.

None of `b.png`, `c.png`, and `d.png` has an `alt` attribute. The word "alt" only appears inside the value of another attribute (`title`, `class`, `data-caption`).

### What actually happened?

Only `a.png` is reported:

```
test.md
  3:1  error  Alternative text for image is required  markdown/require-alt-text

✖ 1 problem (1 error, 0 warnings)
```

`b.png`, `c.png`, and `d.png` are not reported, even though they have no alternative text.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-iqmooju2

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

> Disclosure: I'm a participant of the open source contribution program [OSSCA](https://github.com/eslint-ossca).

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of the open source contribution program [OSSCA](https://github.com/eslint-ossca): confirmed.

I can reproduce this, and it seems that the following helper function also matches the `<img src="b.png" title="an alt">` pattern, resulting in false negatives.

https://github.com/eslint/markdown/blob/4cc3873a631c5d389fb3f468c168018d37ff7fb7/src/rules/require-alt-text.js#L32-L39

A PR is welcome.
